sklearn/utils/validation.py

```diff
@@ -1263,15 +1279,21 @@ def check_scalar(x, name, target_type, *, min_val=None, max_val=None):
     """
 
     if not isinstance(x, target_type):
-        raise TypeError(
-            "`{}` must be an instance of {}, not {}.".format(name, target_type, type(x))
-        )
+        raise TypeError(f"{name} must be an instance of {target_type}, not {type(x)}.")
 
-    if min_val is not None and x < min_val:
-        raise ValueError("`{}`= {}, must be >= {}.".format(name, x, min_val))
+    comparison_operator = operator.le if strictly_greater_min_val else operator.lt
+    if min_val is not None and comparison_operator(x, min_val):
+        raise ValueError(
+            f"{name} == {x}, must be {'>' if strictly_greater_min_val else '>='} "
+            f"{min_val}."
+        )
 
-    if max_val is not None and x > max_val:
-        raise ValueError("`{}`= {}, must be <= {}.".format(name, x, max_val))
+    comparison_operator = operator.ge if strictly_less_max_val else operator.gt
+    if max_val is not None and comparison_operator(x, max_val):
+        raise ValueError(
+            f"{name} == {x}, must be {'<' if strictly_less_max_val else '<='} "
+            f"{max_val}."
+        )
```

Comment:
Maybe returning the converted input value (similarly to `check_array`) is handy?

Reply:
right